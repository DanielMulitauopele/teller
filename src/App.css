* {
  --MainGradient: linear-gradient(
    to right,
    #8065e9,
    #9862da,
    #b760c5,
    #db5db2,
    #fc599f
  );
  --BlueSteel: linear-gradient(to bottom, #7145e3, #5694ee, #6ce9fc);
  --Peachy: linear-gradient(to bottom, #fd746c, #ff9068);
  --BlueGreen: linear-gradient(to bottom, #c2e59c, #64b3f4);
  --Pinkish: linear-gradient(to bottom, #b24592, #f15f79);
  --Descent: linear-gradient(to bottom, #00c6ff, #0072ff);
  --Tranquil: linear-gradient(to bottom, #9862da, #db5db2);
  --AlmostWhite: rgb(230, 230, 230);
  --TellerBG: #303859;
  --TellerBGDark: #232941;
  --TellerPink: #fc599f;
  margin: 0;
  padding: 0;
  outline: none;
  -webkit-overflow-scrolling: touch;
}

@font-face {
  font-family: teller;
  src: url(./Assets/fonts/GeorgeRegular.ttf);
}

.App {
  height: 100vh;
  background-color: var(--TellerBG);
  overflow: hidden;
}

.app-subtle-bg {
  text-align: center;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  color: #232941d5;
  z-index: 0;
}

.app-subtle-bg h1 {
  font-size: 4rem;
  font-family: teller;
}

.app-subtle-bg p {
  font-size: 1.5rem;
}
