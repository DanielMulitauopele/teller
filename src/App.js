import React, { Component } from "react";
import "./App.css";
import Hotdog from "./Components/Hotdog/Hotdog";
import DataCleaner from "./Utils/Cleaners/";
import Search from "./Components/Search/Search";
import { BrowserRouter, Route, Switch } from "react-router-dom";
import Landing from "./Components/Landing/Landing";
import NotesContainer from "./Components/NotesContainer/NotesContainer"
import LoginContainer from "./Components/LoginContainer/LoginContainer";
import Onboarding from "./Components/Onboarding/Onboarding";
import AboutUs from "./Components/AboutUs/AboutUs";

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      favorites: [
        {
          id: 12345,
          name: "No favorites saved",
          price: "0",
          percent_change: "0"
        }
      ],
      abbrevCurrencies: [],
      expandedCurrencies: [],
      userEmail: "",
      news: [],
      loggedIn: false,
      notes: [],
      token: ''
    };
    this.cleaner = new DataCleaner();
  }

  async componentDidMount() {
    const abbrevCurrencies = await this.cleaner.getAbbrevCurrencies();
    const expandedCurrencies = await this.cleaner.getExpandedCurrencies();
    this.setState({ abbrevCurrencies, expandedCurrencies });
  }

  addToFavorites = favorite => {
    const { favorites } = this.state;
    const newFave = { id: Date.now(), ...favorite };
    if (favorites.length < 5) {
      this.setState({ favorites: [newFave, ...favorites] });
    } else if (favorites.length >= 5) {
      this.setState({
        favorites: [newFave, ...favorites.slice(0, 4)]
      });
    }
  };

  removeFromFavorites = id => {
    const filteredFavorites = this.state.favorites.filter(
      favorite => favorite.id !== id
    );
    this.setState({ favorites: filteredFavorites });
  };

  addToNotes = note => {
    const { notes } = this.state;
    const newNote = { id: Date.now(), ...note };
    this.setState({
      notes: [newNote, ...notes]
    });
  };

  removeFromNotes = id => {
    const filteredNotes = this.state.notes.filter(note => note.id !== id);
    this.setState({ notes: filteredNotes });
  };

  logInUser = userEmail => {
    this.setState({ userEmail });
    this.setState({ loggedIn: true });
  };

  setLoginState = () => {
    this.setState({
      loggedIn: !this.state.loggedIn
    });
  };

  displaySearch = async currency => {
    let abbCurr;
    let expCurr;
    const { abbrevCurrencies, expandedCurrencies } = this.state;
    if (currency === "") {
      abbCurr = await this.cleaner.getAbbrevCurrencies();
      expCurr = await this.cleaner.getExpandedCurrencies();
    } else {
      abbCurr = abbrevCurrencies.filter(
        curr =>
          curr.name.toUpperCase().includes(currency.toUpperCase()) ||
          curr.name.toUpperCase() === currency.toUpperCase()
      );
      expCurr = expandedCurrencies.filter(
        curr =>
          curr.name.toUpperCase().includes(currency.toUpperCase()) ||
          curr.name.toUpperCase() === currency.toUpperCase()
      );
    }
    this.setState({
      abbrevCurrencies: abbCurr,
      expandedCurrencies: expCurr
    });
  };

  setFilter = filterCategory => {
    const { abbrevCurrencies, expandedCurrencies } = this.state;
    let sortedAbbrev;
    let sortedExp;
    if (filterCategory === "Rank") {
      sortedAbbrev = abbrevCurrencies.sort((a, b) => a.rank - b.rank);
      sortedExp = expandedCurrencies.sort((a, b) => a.rank - b.rank);
    } else if (filterCategory === "Price") {
      sortedAbbrev = abbrevCurrencies.sort((a, b) => a.price - b.price);
      sortedExp = expandedCurrencies.sort((a, b) => a.price - b.price);
    } else if (filterCategory === "%Change") {
      sortedAbbrev = abbrevCurrencies.sort(
        (a, b) => a.percent_change - b.percent_change
      );
      sortedExp = expandedCurrencies.sort(
        (a, b) => a.percent_change - b.percent_change
      );
    }
    this.setState({
      abbrevCurrencies: sortedAbbrev,
      expandedCurrencies: sortedExp
    });
  };

  storeToken = (token) => {
    this.setState({ token: token.teller_api_token })
    console.log(this.state.token)
  }

  render() {
    const { abbrevCurrencies, favorites, notes } = this.state;
    return (
      <BrowserRouter>
        <div className="App">
          {this.state.loggedIn && <Hotdog setLoginState={this.setLoginState} />}
          {this.state.loggedIn && <Search displaySearch={this.displaySearch} />}
          {this.state.loggedIn && (
            <div className="app-subtle-bg">
              <h1>teller.</h1>
              <p> Your Personal Crypto Analyst </p>
            </div>
          )}
          <Switch>
            <Route
              exact
              path="/home"
              render={() => {
                return (
                  <Landing
                    favorites={favorites}
                    addToFavorites={this.addToFavorites}
                    removeFromFavorites={this.removeFromFavorites}
                    abbrevCurrencies={abbrevCurrencies}
                    setFilter={this.setFilter}
                  />
                );
              }}
            />
            )}}
            <Route
              exact
              path="/"
              render={() => {
                return <LoginContainer 
                          loggedIn={this.setLoginState}
                          storeToken={this.storeToken} />;
              }}
            />
            <Route
              exact
              path="/onboarding"
              render={() => {
                return <Onboarding />;
              }}
            />
            <Route
              exact
              path="/notes"
              render={() => {
                return (
                  <NotesContainer
                    notes={notes}
                    addToNotes={this.addToNotes}
                    removeFromNotes={this.removeFromNotes}
                  />
                );
              }}
            />
            <Route
              exact
              path="/about"
              render={() => {
                return <AboutUs />;
              }}
            />
            <Route 
              exact
              path="/notes"
              render={() => {
                return (
                  <NotesContainer 
                    notes={notes}
                    addToNotes={this.addToNotes}
                    removeFromNotes={this.removeFromNotes}
                  />
                )
              }}
            />
          </Switch>
        </div>
      </BrowserRouter>
    );
  }
}

export default App;
