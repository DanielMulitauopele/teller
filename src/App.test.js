import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow } from 'enzyme'
import { mockFavorites } from './Utils/MockData/mockFavorites'
import { mockNotes } from './Utils/MockData/mockNotes'
import { mockAbbrevCurrencies } from './Utils/MockData/mockAbbrevCurrencies'
import { mockExpandedCurrencies } from './Utils/MockData/mockExpandedCurrencies'

let wrapper

beforeEach(() => {
  wrapper = shallow(<App />)
})

describe('App Component', () => {
  xit('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should start with the correct default state', () => {
    expect(wrapper.state().favorites.length).toEqual(1)
    expect(wrapper.state().abbrevCurrencies).toEqual([])
    expect(wrapper.state().expandedCurrencies).toEqual([])
    expect(wrapper.state().userEmail).toEqual("")
    expect(wrapper.state().news).toEqual([])
    expect(wrapper.state().notes).toEqual([])
  })

  describe('componentDidMount function', () => {
    xit('should set state with currencies', async () => {

    })
  })

  describe('addToFavorites function', () => {
    it('should add currency to favorites array', () => {
      wrapper.instance().addToFavorites(mockFavorites[0])
      expect(wrapper.state().favorites.length).toEqual(2)
    })

    it('should only allow 5 currencies to be added to favorites', () => {
      const expected = {
        name: "No favorites saved", 
        price: "0",
        percent_change: "0"
      }
      wrapper.instance().addToFavorites(mockFavorites[0])
      wrapper.instance().addToFavorites(mockFavorites[1])
      wrapper.instance().addToFavorites(mockFavorites[2])
      wrapper.instance().addToFavorites(mockFavorites[3])
      wrapper.instance().addToFavorites(mockFavorites[4])
      expect(wrapper.state().favorites).not.toContain(expected)
    })
  })

  describe('removeFromFavorites function', () => {
    it('should remove the favorite with the correct id from state', () => {
      wrapper.instance().addToFavorites(mockFavorites[0])
      expect(wrapper.state().favorites.length).toEqual(2)
      wrapper.instance().removeFromFavorites(wrapper.state().favorites[0].id)
      expect(wrapper.state().favorites.length).toEqual(1)
    })
  })

  describe('addToNotes function', () => {
    it('should add a note to the notes array', () => {
      wrapper.instance().addToNotes(mockNotes[0])
      expect(wrapper.state().notes.length).toEqual(1)
    })
  })

  describe('removeFromNotes function', () => {
    it('should remove the note with the correct id from state', () => {
      wrapper.instance().addToNotes(mockNotes[0])
      expect(wrapper.state().notes.length).toEqual(1)
      wrapper.instance().removeFromNotes(wrapper.state().notes[0].id)
      expect(wrapper.state().notes.length).toEqual(0)
    })
  })

  describe('loginUser function', () => {
    it('should set state with user email', () => {
      const userEmail = "bitcoinfanatic@gmail.com"
      wrapper.instance().logInUser(userEmail)
      expect(wrapper.state().userEmail).toEqual(userEmail)
    })
  })

  describe('displaySearch function', () => {
    xit('should set state with filtered coins that match the argument', () => {

    })
    
    xit('should call getAbbrevCurrencies if search field is empty', () => {
      wrapper.instance().displaySearch("")


    })

    xit('should call getExpandedCurrencies if search field is empty', () => {

    })
  })

  describe('setFilter function', () => {
    xit('should sort the currencies by rank if category is rank', () => {

    })

    xit('should sort the currencies by price if category is price', () => {

    })

    xit('should sort the currencies by %change if category is %change', () => {

    })
  })
})
