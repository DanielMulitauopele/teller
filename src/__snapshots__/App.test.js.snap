// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="App"
      >
        <Switch>
          <Route
            exact={true}
            path="/home"
            render={[Function]}
          />
          )}}
          <Route
            exact={true}
            path="/"
            render={[Function]}
          />
          <Route
            exact={true}
            path="/onboarding"
            render={[Function]}
          />
          <Route
            exact={true}
            path="/notes"
            render={[Function]}
          />
          <Route
            exact={true}
            path="/about"
            render={[Function]}
          />
          <Route
            exact={true}
            path="/notes"
            render={[Function]}
          />
        </Switch>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          false,
          false,
          <Switch>
            <Route
              exact={true}
              path="/home"
              render={[Function]}
            />
            )}}
            <Route
              exact={true}
              path="/"
              render={[Function]}
            />
            <Route
              exact={true}
              path="/onboarding"
              render={[Function]}
            />
            <Route
              exact={true}
              path="/notes"
              render={[Function]}
            />
            <Route
              exact={true}
              path="/about"
              render={[Function]}
            />
            <Route
              exact={true}
              path="/notes"
              render={[Function]}
            />
          </Switch>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        false,
        false,
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                exact={true}
                path="/home"
                render={[Function]}
              />,
              ")}}",
              <Route
                exact={true}
                path="/"
                render={[Function]}
              />,
              <Route
                exact={true}
                path="/onboarding"
                render={[Function]}
              />,
              <Route
                exact={true}
                path="/notes"
                render={[Function]}
              />,
              <Route
                exact={true}
                path="/about"
                render={[Function]}
              />,
              <Route
                exact={true}
                path="/notes"
                render={[Function]}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/home",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            ")}}",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/onboarding",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/notes",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/about",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/notes",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="App"
        >
          <Switch>
            <Route
              exact={true}
              path="/home"
              render={[Function]}
            />
            )}}
            <Route
              exact={true}
              path="/"
              render={[Function]}
            />
            <Route
              exact={true}
              path="/onboarding"
              render={[Function]}
            />
            <Route
              exact={true}
              path="/notes"
              render={[Function]}
            />
            <Route
              exact={true}
              path="/about"
              render={[Function]}
            />
            <Route
              exact={true}
              path="/notes"
              render={[Function]}
            />
          </Switch>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            false,
            false,
            false,
            <Switch>
              <Route
                exact={true}
                path="/home"
                render={[Function]}
              />
              )}}
              <Route
                exact={true}
                path="/"
                render={[Function]}
              />
              <Route
                exact={true}
                path="/onboarding"
                render={[Function]}
              />
              <Route
                exact={true}
                path="/notes"
                render={[Function]}
              />
              <Route
                exact={true}
                path="/about"
                render={[Function]}
              />
              <Route
                exact={true}
                path="/notes"
                render={[Function]}
              />
            </Switch>,
          ],
          "className": "App",
        },
        "ref": null,
        "rendered": Array [
          false,
          false,
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Route
                  exact={true}
                  path="/home"
                  render={[Function]}
                />,
                ")}}",
                <Route
                  exact={true}
                  path="/"
                  render={[Function]}
                />,
                <Route
                  exact={true}
                  path="/onboarding"
                  render={[Function]}
                />,
                <Route
                  exact={true}
                  path="/notes"
                  render={[Function]}
                />,
                <Route
                  exact={true}
                  path="/about"
                  render={[Function]}
                />,
                <Route
                  exact={true}
                  path="/notes"
                  render={[Function]}
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "exact": true,
                  "path": "/home",
                  "render": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              ")}}",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "exact": true,
                  "path": "/",
                  "render": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "exact": true,
                  "path": "/onboarding",
                  "render": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "exact": true,
                  "path": "/notes",
                  "render": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "exact": true,
                  "path": "/about",
                  "render": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "exact": true,
                  "path": "/notes",
                  "render": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
